### PR TITLE
Add VC SDK to Wallet Library.

### PR DESCRIPTION
**Pull request by nithyaganeshng**
- Move VC SDK into Wallet Library.
- Make the VC SDK classes internal.
- Removed git submodule for VC SDK.
- Removed gradle tasks added to include VC SDK as part of Wallet Library.
- Moved VC SDK tests into Wallet Library.

**Solution:**
Used library A to complete scenario X with quality and style.

**Validation:**

- Verified all the VC SDK and Wallet Library tests including unit tests and instrumented tests.
- Published wallet library package locally and verified the flow in demo app.

**Type of change:**
- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [x] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/IssuanceResponseFormatter.kt [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+package com.microsoft.did.sdk.credential.service.protectors
+
+import android.util.Base64
+import com.microsoft.did.sdk.credential.service.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/IssuanceResponseFormatter.kt]: diff from the repo shows this was:
```kotlin
import com.microsoft.did.sdk.credential.service.IssuanceResponse
import com.microsoft.did.sdk.credential.service.RequestedAccessTokenMap
import com.microsoft.did.sdk.credential.service.RequestedIdTokenMap
import com.microsoft.did.sdk.credential.service.RequestedSelfAttestedClaimMap
import com.microsoft.did.sdk.credential.service.RequestedVcMap
```
Do we have a style guide that dictates this change?

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/IssuanceResponseFormatter.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import com.microsoft.did.sdk.credential.service.models.oidc.AttestationClaimModel
+import com.microsoft.did.sdk.credential.service.models.oidc.IssuanceResponseClaims
+import com.microsoft.did.sdk.crypto.CryptoOperations
+import com.microsoft.did.sdk.crypto.DigestAlgorithm
+import com.microsoft.did.sdk.crypto.keyStore.EncryptedKeyStore
+import com.microsoft.did.sdk.identifier.models.Identifier
+import kotlinx.serialization.json.Json
+import java.util.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/IssuanceResponseFormatter.kt]: Diff also shows this was explicitly `java.util.UUID` (across a number of files).

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/IssuanceResponseFormatter.kt (hunk 1 above)
Comment (nithyaganeshng) [reply walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/IssuanceResponseFormatter.kt]: These changes have been automatically made by IDE when I moved the files here. I will check the settings on mine.

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/validators/OidcPresentationRequestValidator.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package com.microsoft.did.sdk.credential.service.validators
+
+import com.microsoft.did.sdk.credential.service.PresentationRequest
+import com.microsoft.did.sdk.credential.service.models.oidc.PinDetails
+import com.microsoft.did.sdk.credential.service.models.oidc.Registration
+import com.microsoft.did.sdk.util.Constants
+import com.microsoft.did.sdk.util.Constants.MILLISECONDS_IN_A_SECOND
+import com.microsoft.did.sdk.util.Constants.SECONDS_IN_A_MINUTE
+import com.microsoft.did.sdk.util.controlflow.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/validators/OidcPresentationRequestValidator.kt]: Changed from:
```kotlin
import com.microsoft.did.sdk.util.controlflow.InvalidPinDetailsException
import com.microsoft.did.sdk.util.controlflow.InvalidResponseModeException
import com.microsoft.did.sdk.util.controlflow.InvalidResponseTypeException
import com.microsoft.did.sdk.util.controlflow.InvalidScopeException
import com.microsoft.did.sdk.util.controlflow.MissingInputInRequestException
import com.microsoft.did.sdk.util.controlflow.SubjectIdentifierTypeNotSupported
import com.microsoft.did.sdk.util.controlflow.VpFormatNotSupported
```
?

File: walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/Algorithms.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved
+
+package com.microsoft.did.sdk.crypto
+
+import com.microsoft.did.sdk.crypto.spi.EcPairwisePrivateKeySpec
+import com.microsoft.did.sdk.crypto.spi.EcPairwisePublicKeySpec
+import com.nimbusds.jose.jwk.Curve
+import java.math.BigInteger
+import java.security.spec.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/Algorithms.kt]:
```kotlin
import java.security.spec.AlgorithmParameterSpec
import java.security.spec.ECPoint
import java.security.spec.ECPrivateKeySpec
import java.security.spec.ECPublicKeySpec
import java.security.spec.KeySpec
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/CryptoOperations.kt [hunk 5]
```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+package com.microsoft.did.sdk.crypto
+
+import com.microsoft.did.sdk.util.Constants.SEED_BYTES
+import java.security.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/CryptoOperations.kt]:
```
import java.security.Key
import java.security.KeyFactory
import java.security.KeyPair
import java.security.KeyPairGenerator
import java.security.MessageDigest
import java.security.PrivateKey
import java.security.PublicKey
import java.security.SecureRandom
import java.security.Signature
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jwe/JweToken.kt [hunk 6]
```diff
@@ -0,0 +1,117 @@
+package com.microsoft.did.sdk.crypto.protocols.jose.jwe
+
+import com.microsoft.did.sdk.crypto.keyStore.EncryptedKeyStore
+import com.microsoft.did.sdk.crypto.protocols.jose.JwaCryptoHelper
+import com.microsoft.did.sdk.util.controlflow.AlgorithmException
+import com.microsoft.did.sdk.util.controlflow.FailedDecryptException
+import com.microsoft.did.sdk.util.controlflow.KeyException
+import com.nimbusds.jose.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jwe/JweToken.kt]:
```kotlin
import com.nimbusds.jose.EncryptionMethod
import com.nimbusds.jose.JOSEException
import com.nimbusds.jose.JWEAlgorithm
import com.nimbusds.jose.JWEDecrypter
import com.nimbusds.jose.JWEEncrypter
import com.nimbusds.jose.JWEHeader
import com.nimbusds.jose.JWEObject
import com.nimbusds.jose.Payload
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jwe/JweToken.kt [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+import com.nimbusds.jose.crypto.ECDHEncrypter
+import com.nimbusds.jose.crypto.PasswordBasedEncrypter
+import com.nimbusds.jose.crypto.RSAEncrypter
+import com.nimbusds.jose.crypto.X25519Encrypter
+import com.nimbusds.jose.crypto.factories.DefaultJWEDecrypterFactory
+import com.nimbusds.jose.jwk.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jwe/JweToken.kt]:
```kotlin
import com.nimbusds.jose.jwk.ECKey
import com.nimbusds.jose.jwk.JWK
import com.nimbusds.jose.jwk.KeyConverter
import com.nimbusds.jose.jwk.OctetKeyPair
import com.nimbusds.jose.jwk.OctetSequenceKey
import com.nimbusds.jose.jwk.RSAKey
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/serialization/JwkSerializer.kt [hunk 8]
```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved
+
+package com.microsoft.did.sdk.crypto.protocols.jose.serialization
+
+import com.nimbusds.jose.jwk.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/serialization/JwkSerializer.kt]:
```
import com.nimbusds.jose.jwk.ECKey
import com.nimbusds.jose.jwk.JWK
import com.nimbusds.jose.jwk.OctetKeyPair
import com.nimbusds.jose.jwk.OctetSequenceKey
import com.nimbusds.jose.jwk.RSAKey
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/BaseNetworkOperation.kt [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+package com.microsoft.did.sdk.datasource.network
+
+import com.microsoft.did.sdk.util.Constants.CORRELATION_VECTOR_HEADER
+import com.microsoft.did.sdk.util.Constants.REQUEST_ID_HEADER
+import com.microsoft.did.sdk.util.NetworkErrorParser
+import com.microsoft.did.sdk.util.controlflow.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/BaseNetworkOperation.kt]:
```
import com.microsoft.did.sdk.util.controlflow.ClientException
import com.microsoft.did.sdk.util.controlflow.ForbiddenException
import com.microsoft.did.sdk.util.controlflow.LocalNetworkException
import com.microsoft.did.sdk.util.controlflow.NetworkException
import com.microsoft.did.sdk.util.controlflow.NotFoundException
import com.microsoft.did.sdk.util.controlflow.RedirectException
import com.microsoft.did.sdk.util.controlflow.Result
import com.microsoft.did.sdk.util.controlflow.ServiceUnreachableException
import com.microsoft.did.sdk.util.controlflow.UnauthorizedException
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/apis/IssuanceApis.kt [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+package com.microsoft.did.sdk.datasource.network.apis
+
+import com.microsoft.did.sdk.credential.service.models.serviceResponses.ContractServiceResponse
+import com.microsoft.did.sdk.credential.service.models.serviceResponses.IssuanceServiceResponse
+import retrofit2.Response
+import retrofit2.http.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/apis/IssuanceApis.kt]:
```
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.POST
import retrofit2.http.Url
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/apis/PresentationApis.kt [hunk 11]
```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+package com.microsoft.did.sdk.datasource.network.apis
+
+import retrofit2.Response
+import retrofit2.http.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/apis/PresentationApis.kt]:
```
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.POST
import retrofit2.http.Url
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+ *--------------------------------------------------------------------------------------------*/
+
+package com.microsoft.did.sdk.datasource.network.credentialOperations
+
+import com.microsoft.did.sdk.credential.service.models.oidc.PresentationRequestContent
+import com.microsoft.did.sdk.credential.service.validators.JwtValidator
+import com.microsoft.did.sdk.crypto.protocols.jose.jws.JwsToken
+import com.microsoft.did.sdk.datasource.network.GetNetworkOperation
+import com.microsoft.did.sdk.datasource.network.apis.ApiProvider
+import com.microsoft.did.sdk.util.controlflow.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/FetchPresentationRequestNetworkOperation.kt]:
```
import com.microsoft.did.sdk.util.controlflow.DidInHeaderAndPayloadNotMatching
import com.microsoft.did.sdk.util.controlflow.ExpiredTokenException
import com.microsoft.did.sdk.util.controlflow.InvalidSignatureException
import com.microsoft.did.sdk.util.controlflow.NotFoundException
import com.microsoft.did.sdk.util.controlflow.PresentationException
import com.microsoft.did.sdk.util.controlflow.Result
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/SendVerifiableCredentialIssuanceRequestNetworkOperation.kt [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+import com.microsoft.did.sdk.credential.models.VerifiableCredential
+import com.microsoft.did.sdk.credential.models.VerifiableCredentialContent
+import com.microsoft.did.sdk.credential.service.models.serviceResponses.IssuanceServiceResponse
+import com.microsoft.did.sdk.credential.service.validators.JwtValidator
+import com.microsoft.did.sdk.crypto.protocols.jose.jws.JwsToken
+import com.microsoft.did.sdk.datasource.network.PostNetworkOperation
+import com.microsoft.did.sdk.datasource.network.apis.ApiProvider
+import com.microsoft.did.sdk.util.Constants
+import com.microsoft.did.sdk.util.controlflow.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/datasource/network/credentialOperations/SendVerifiableCredentialIssuanceRequestNetworkOperation.kt]:
```
import com.microsoft.did.sdk.util.controlflow.ForbiddenException
import com.microsoft.did.sdk.util.controlflow.InvalidPinException
import com.microsoft.did.sdk.util.controlflow.InvalidSignatureException
import com.microsoft.did.sdk.util.controlflow.IssuanceException
import com.microsoft.did.sdk.util.controlflow.Result
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/util/DifWordList.kt [hunk 14]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved
+
+package com.microsoft.did.sdk.util
+
+import android.content.Context
+import android.content.res.AssetManager
+import java.io.InputStreamReader
+import java.security.SecureRandom
+import java.util.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/util/DifWordList.kt]:
```
import java.util.Locale
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+import com.microsoft.did.sdk.identifier.models.Identifier
+import com.microsoft.did.sdk.identifier.models.payload.RegistrationPayload
+import com.microsoft.did.sdk.identifier.models.payload.SuffixData
+import com.microsoft.did.sdk.util.Constants
+import com.microsoft.did.sdk.util.Constants.AES_KEY
+import com.nimbusds.jose.jwk.JWK
+import com.nimbusds.jose.jwk.KeyUse
+import kotlinx.serialization.json.Json
+import org.erdtman.jcs.JsonCanonicalizer
+import java.util.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt]:
```
import java.util.UUID
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/validators/OidcPresentationRequestValidator.kt [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/validators/OidcPresentationRequestValidator.kt]:
```
import java.util.Date
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/VerifiablePresentationFormatter.kt [hunk 17]
```diff
@@ -0,0 +1,82 @@
+package com.microsoft.did.sdk.credential.service.protectors
+
+import com.microsoft.did.sdk.credential.models.VerifiableCredential
+import com.microsoft.did.sdk.credential.service.models.verifiablePresentation.VerifiablePresentationContent
+import com.microsoft.did.sdk.credential.service.models.verifiablePresentation.VerifiablePresentationDescriptor
+import com.microsoft.did.sdk.identifier.models.Identifier
+import com.microsoft.did.sdk.util.Constants
+import kotlinx.serialization.json.Json
+import java.util.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/VerifiablePresentationFormatter.kt]:
```
import java.util.UUID
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/RevocationResponseFormatter.kt [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+
+package com.microsoft.did.sdk.credential.service.protectors
+
+import com.microsoft.did.sdk.credential.service.models.RevocationRequest
+import com.microsoft.did.sdk.credential.service.models.oidc.RevocationResponseClaims
+import com.microsoft.did.sdk.crypto.keyStore.EncryptedKeyStore
+import com.microsoft.did.sdk.identifier.models.Identifier
+import com.microsoft.did.sdk.util.Constants
+import kotlinx.serialization.json.Json
+import java.util.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/RevocationResponseFormatter.kt]:
```
import java.util.UUID
```

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/PresentationResponseFormatter.kt [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+import com.microsoft.did.sdk.credential.service.RequestedVcPresentationSubmissionMap
+import com.microsoft.did.sdk.credential.service.models.oidc.PresentationResponseClaims
+import com.microsoft.did.sdk.credential.service.models.oidc.VpTokenInResponse
+import com.microsoft.did.sdk.credential.service.models.presentationexchange.PresentationSubmission
+import com.microsoft.did.sdk.credential.service.models.presentationexchange.PresentationSubmissionDescriptor
+import com.microsoft.did.sdk.identifier.models.Identifier
+import com.microsoft.did.sdk.util.Constants
+import com.microsoft.did.sdk.util.Constants.DEFAULT_VP_EXPIRATION_IN_SECONDS
+import kotlinx.serialization.json.Json
+import java.util.*
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/PresentationResponseFormatter.kt]:
```
import java.util.UUID
```

**Review by logangirvin — DISMISSED** [review]
I could only find changes to `internal` or import statements that seem automatic/benign.

File: walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/IssuanceResponseFormatter.kt (hunk 1 above)
Comment (nithyaganeshng) [reply walletlibrary/src/main/java/com/microsoft/did/sdk/credential/service/protectors/IssuanceResponseFormatter.kt]: reverted the styling changes.

**Review by logangirvin — APPROVED** [review]

**Pull request merged**